src/components/tooltip/tooltip.js

```diff
@@ -0,0 +1,198 @@
+    clearTimeout(this.leaveTimer);
+  }
+
+  handleEnter = event => {
+    const childrenProps = this.props.children.props;
+
+    // Remove the title ahead of time.
+    // We don't want to wait for the next render commit.
+    // We would risk displaying two tooltips at the same time (native + this one).
+    if (this.childrenRef && typeof this.childrenRef === 'function') {
+      this.childrenRef.setAttribute('title', '');
+    }
+
+    if (event.type === 'mouseover' && childrenProps.onMouseOver) {
+      childrenProps.onMouseOver(event);
+    }
+
+    if (event.type === 'focus' && childrenProps.onFocus) {
+      childrenProps.onFocus(event);
+    }
+
+    if (!this.state.open && !this.isControlled) {
+      this.setState({
+        open: true,
+      });
+    }
+
+    if (this.props.onOpen) {
+      this.props.onOpen(event);
+    }
+
+    event.preventDefault();
+    event.stopPropagation();
+  };
+
+  handleLeave = event => {
+    const childrenProps = this.props.children.props;
+    clearTimeout(this.leaveTimer);
+
+    if (event.type === 'mouseleave' && childrenProps.onMouseLeave) {
+      childrenProps.onMouseLeave(event);
+    }
+
+    if (event.type === 'blur' && childrenProps.onBlur) {
+      childrenProps.onBlur(event);
+    }
+
+    if (this.props.leaveDelay) {
+      this.leaveTimer = setTimeout(() => {
+        this.handleClose();
+      }, this.props.leaveDelay);
+    } else {
+      this.handleClose();
+    }
+  };
+
+  handleClose = event => {
+    if (!this.isControlled) {
+      this.setState({
+        open: false,
+      });
+    }
+    if (this.props.onClose) {
+      this.props.onClose(event);
+    }
+  };
+
+  setChildrenRef = ref => {
+    this.childrenRef = ref;
+  };
+
+  isControlled = this.props.isOpen != null;
+
+  render() {
+    const open = this.isControlled ? this.props.isOpen : this.state.open;
+
+    const childrenProps = {
+      'aria-describedby': open ? this.state.id : null,
+      // for seo and accessibility, we add the tooltip's title
+      // as a native title when the title is hidden
+      title:
+        !open && typeof this.props.title === 'string' ? this.props.title : null,
+      ...this.props.children.props,
+    };
+
+    const eventListeners = {
+      onMouseOver: this.handleEnter,
+      onMouseLeave: this.handleLeave,
+      onFocus: this.handleEnter,
+      onBlur: this.handleLeave,
+    };
+
+    const WrapperComponent = this.props.components.WrapperComponent || Wrapper;
+
+    return (
+      <Manager>
+        <Reference innerRef={this.setChildrenRef}>
+          {({ ref }) => (
+            <WrapperComponent {...eventListeners}>
+              <RootRef rootRef={ref}>
+                {React.cloneElement(this.props.children, { ...childrenProps })}
+              </RootRef>
+            </WrapperComponent>
+          )}
+        </Reference>
+        {open && (
+          <Popper placement={this.props.placement}>
+            {({ ref, style, placement }) => (
+              <div
```

Comment:
We wanted to extract this into a `<TooltipContent>` or something, in order to run VRT on it, right?

Reply:
I thought about that but it seems a bit weird since then we would need to export that, since the VRT uses the bundled ui-kit...

Reply:
Hmm right...that's unfortunate 😑